src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -160,7 +161,12 @@ public IReadOnlyList<TagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotS
                         // tags. Only dealing with the dataTagSpans if the count is 1 because we do not see a multi-buffer case
                         // occurring
                         var dataTagSpans = tag.Span.GetSpans(snapshot);
-                        if (dataTagSpans.Count == 1)
+
+                        // Checks to see if location and hint name are already present in the cache.
```

Comment:
Why are we getting duplicates in the first place? It doesn't surprise me that we'd have to filter out duplicates at some level of the system, but we're sure we're not getting duplicates here because we're not clearing out tags or something else is returning tags outside the spans we're asking for, etc.?

Reply:
it should be easy to debug through as well to see what is causing dupes here.

Reply:
As for why we're getting the duplicates, unsure since there are not any dupes at the point at which the InlineHintDataTags are being created, in the InlineHintDataTaggerProvider.

Reply:
Please investigate this.  We need to know what is going on here to determine if this is the right fix.  

> unsure since there are not any dupes at the point at which the InlineHintDataTags are being created, in the InlineHintDataTaggerProvider.

They're coming from somewhere :)